jenkins-plugin/src/test/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepTest.java

```diff
@@ -96,6 +106,47 @@ public void testDisableAllPublishers() throws Exception {
         assertFingerprintDoesNotExist(COMMONS_LANG3_FINGERPRINT);
     }
 
+    // the jdk path is configured in Dockerfile
+    private static Stream<Arguments> jdkMapProvider() {
+        return Stream.of(
+                arguments("jdk8", "/opt/java/jdk8"),
+                arguments("jdk11", "/opt/java/jdk11")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("jdkMapProvider")
+    @Issue("JENKINS-71949")
+    public void tesWithDifferentJavasForBuild(String jdkName, String jdkPath) throws Exception {
+        loadMonoDependencyMavenProjectInGitRepo(this.gitRepoRule);
+        String gitRepoPath = this.gitRepoRule.toString();
+        javaGitContainerRule.copyFileToContainer(MountableFile.forHostPath(gitRepoPath), "/tmp/gitrepo");
```

Comment:
I fear `forHostPath` will not work on CI because the source (`gitRepoPath`) will be on the agent running the build, which is itself a Docker container, not the bare-metal host running this container

Reply:
nah this will definitely work. it's just a matter of files permissions. as it's some docker on docker on docker etc.... somewhere in the chain it's not rootless and so the files were copied a root only. 
the chmod the line below (https://github.com/jenkinsci/pipeline-maven-plugin/pull/703/files#diff-0227c66ed42b4533f30dfb8dd90631e73c2b206980eeb2af20db7a2edd19388bR124) will fix the problem.

Reply:
see build 11 is green https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fpipeline-maven-plugin/detail/PR-703/11/pipeline

Reply:
ok, nice !